### ISSUE TITLE
Subscription - Billed twice

### ISSUE THREAD

**Issue by gzumaglini** [paypal/PayPal-PHP-SDK#1209]
### General information

* SDK/Library version: 1.13.0
* Environment: Production
* `PayPal-Debug-ID` values: -
<img width="424" alt="billing" src="https://user-images.githubusercontent.com/1321431/48722882-77a44700-ec25-11e8-80a3-70694190b2fd.png">

* Language, language version, and OS: PHP 5.6.36-1+ubuntu14.04.1+deb.sury.org+1

### Issue description

I have setup a subscription flow using mostly code from the examples.

As i want to bill the user immediately i have used the workaround with the setup_fee as mentioned here: https://github.com/paypal/PayPal-PHP-SDK/issues/960

On the Sandbox everything works fine, the user gets billed immediately (once, via setup_fee) and when the billing cycle happens (for example on the next day).

But on our production site the user immediately gets billed twice. In our webhook log i can see that 2 payments get created.

This is how i setup the plan (only relevant plans)

`
            $plan = new Plan();
            $plan->setName(...
                ->setType('infinite');

           $amount = 1.10;

            $paymentDefinition = new PaymentDefinition();
            $paymentDefinition->setName(...
                ->setType('REGULAR')
                ->setFrequency('monthly)
                ->setFrequencyInterval(1)
                ->setCycles("0")
                ->setAmount(new Currency(array('value' => floatval($amount), 'currency' => 'EUR')));

            $merchantPreferences = new MerchantPreferences();
            $merchantPreferences->...
                ->setAutoBillAmount("no")
                ->setInitialFailAmountAction("CANCEL")
                ->setSetupFee(new Currency(array('value' => floatval($amount), 'currency' => 'EUR')))
                ->setMaxFailAttempts("0");

            $plan->setPaymentDefinitions(array($paymentDefinition));
            $plan->setMerchantPreferences($merchantPreferences);

            $createdPlan = $plan->create($apiContext);

            // Activate plan
            $patch = new Patch();
            $value = new PayPalModel('{"state":"ACTIVE"}');
            $patch->setOp('replace')
                ->setPath('/')
                ->setValue($value);
            $patchRequest = new PatchRequest();
            $patchRequest->addPatch($patch);

            $createdPlan->update($patchRequest, $apiContext);
            $createdPlan = Plan::get($createdPlan->getId(), $apiContext);

            // Create agreement
            $dateNextRun = '2018-12-19'; // In this example: today is 2018-11-19, so first recurring billing should occur next month
            $now = new \DateTime('now', new \DateTimeZone('Europe/Berlin'));
            $dt = new \DateTime($dateNextRun . ' ' . $now->format('H:i:s'), new \DateTimeZone('Europe/Berlin'));

            $agreement = new Agreement();
            $agreement->setName...
                ->setStartDate(date("Y-m-d\TH:i:s\Z", strtotime($dt->format('Y-m-d H:i:s'))));

            $plan = new Plan();
            $plan->setId($createdPlan->getId());
            $agreement->setPlan($plan);

            $payer = new Payer();
            $payer->setPaymentMethod('paypal');
            $agreement->setPayer($payer);

            $agreement = $agreement->create($apiContext);`

Can you give me some clues where i'm doing something wrong?

**Comment by gzumaglini** [comment]
It turns out that the User doesn't really get billed twice.

What happens is that PayPal sends the Webhook for PAYMENT.SALE.COMPLETED twice for the same sale.

Anyone can tell me why that happens?

**Comment by kslimani** [comment]
@gzumaglini : does the resource state equals `completed` in the two events ?

**Comment by gzumaglini** [comment]
yes, both webhook events have resource state "completed"

**Comment by manuman94** [comment]
Yes! I am experiencing the same issue. 

In production environment PayPal sends two different webhook events (with different event IDs) but pointing to the same resource and type. Both are PAYMENT.SALE.COMPLETED and they are pointing to the same "sale" resource. The payload is exactly the same in both notifications except for the event ID.

It seems to be a PayPal error because I haven't changed the PayPal implementation for several months and this issue is coming up recently (since 18th Nov).

For the moment we are ignoring this event when our system has received already the first, so it's not a real problem but the better would be that this event is fired once.

Greetings.

**Comment by manuman94** [comment]
Here is an example of what we receive in both notifications:

First:
```
{
	"id": "XXXXXXXXXXXXXXXXXX",
	"create_time": "2018-11-21T20:01:40.983Z",
	"resource_type": "sale",
	"event_type": "PAYMENT.SALE.COMPLETED",
	"summary": "Payment completed for EUR 34.0 EUR",
	"resource": {
		"billing_agreement_id": "XXXXXXXX",
		"amount": {
			"total": "34.00",
			"currency": "EUR",
			"details": {
				"subtotal": "34.00"
			}
		},
		"payment_mode": "INSTANT_TRANSFER",
		"update_time": "2018-11-21T20:01:00Z",
		"create_time": "2018-11-21T20:01:00Z",
		"protection_eligibility_type": "ITEM_NOT_RECEIVED_ELIGIBLE,UNAUTHORIZED_PAYMENT_ELIGIBLE",
		"transaction_fee": {
			"value": "1.51",
			"currency": "EUR"
		},
		"protection_eligibility": "ELIGIBLE",
		"links": [
			{
				"href": "https://api.paypal.com/v1/payments/sale/XXXXXXXXXXXXXXXXX",
				"rel": "self",
				"method": "GET"
			},
			{
				"href": "https://api.paypal.com/v1/payments/sale/XXXXXXXXXXXXXXXXX/refund",
				"rel": "refund",
				"method": "POST"
			}
		],
		"id": "XXXXXXXXXXXXXXXXX",
		"state": "completed",
		"invoice_number": ""
	},
	"status": "SUCCESS",
	"transmissions": [
		{
			// SOME HIDDEN DATA HERE
                }
	],
	"links": [
		{
			"href": "https://api.paypal.com/v1/notifications/webhooks-events/XXXXXXXXXXXXXXXXX",
			"rel": "self",
			"method": "GET",
			"encType": "application/json"
		},
		{
			"href": "https://api.paypal.com/v1/notifications/webhooks-events/XXXXXXXXXXXXXXXXX/resend",
			"rel": "resend",
			"method": "POST",
			"encType": "application/json"
		}
	],
	"event_version": "1.0"
}
```

Second notification:

```
{
	"id": "XXXXXXXXXXXXXXXXX",
	"create_time": "2018-11-21T20:01:10.395Z",
	"resource_type": "sale",
	"event_type": "PAYMENT.SALE.COMPLETED",
	"summary": "Payment completed for EUR 34.00 EUR",
	"resource": {
		"billing_agreement_id": "XXXXXXXXXXXXXXXXX",
		"update_time": "2018-11-21T12:01:00Z",
		"payment_mode": "INSTANT_TRANSFER",
		"amount": {
			"total": "34.00",
			"currency": "EUR"
		},
		"create_time": "2018-11-21T12:01:00Z",
		"protection_eligibility_type": "ITEM_NOT_RECEIVED_ELIGIBLE,UNAUTHORIZED_PAYMENT_ELIGIBLE",
		"transaction_fee": {
			"value": "1.51",
			"currency": "EUR"
		},
		"links": [
			{
				"href": "https://https://api.paypal.com/v1/payments/sale/XXXXXXXXXXXXXXXXX",
				"rel": "self",
				"method": "GET"
			},
			{
				"href": "https://https://api.paypal.com/v1/payments/sale/XXXXXXXXXXXXXXXXX/refund",
				"rel": "refund",
				"method": "POST"
			}
		],
		"protection_eligibility": "ELIGIBLE",
		"state": "completed",
		"id": "XXXXXXXXXXXXXXXXX"
	},
	"status": "SUCCESS",
	"transmissions": [
		{
			// SOME HIDDEN DATA HERE
		}
	],
	"links": [
		{
			"href": "https://api.paypal.com/v1/notifications/webhooks-events/XXXXXXXXXXXXXXXXX",
			"rel": "self",
			"method": "GET",
			"encType": "application/json"
		},
		{
			"href": "https://api.paypal.com/v1/notifications/webhooks-events/XXXXXXXXXXXXXXXXX/resend",
			"rel": "resend",
			"method": "POST",
			"encType": "application/json"
		}
	],
	"event_version": "1.0"
}
```

Some data is hidden for security. The resource in both messages is the same and the only difference is the Event ID (the first id field).

Greetings.

**Comment by kslimani** [comment]
I confirm the issue, we noted the same bahaviour.
_(workaround is to use IPN)_

**Comment by younkim** [comment]
I'm receiving two on _a few_ `PAYMENT.SALE.COMPLETE`. It's not clear why. Payloads are essentially the same except for a few unimportant fields. I'm using both rest and NVP webhooks, but only have `PAYMENT.SALE.COMPLETE` turned on for rest. I filed a technical support ticket this a while back, but there hasn't been a response. This usually means it's a bug and that they're looking into it from my experience working with them. I'll post an update if I hear back from them.

**Comment by razorsharpshady** [comment]
Facing same issue,  On mothly billing cycle(recurring payment) PAYMENT.SALE.COMPLETED triggered twice.
But One Time Payment webhook triggers only once.

**Comment by prakash-gangadharan** [comment]
Hi all, if this still an issue, can you help with the recent `PayPal-Debug-ID`
